dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/gist/GistBuilder.java

```diff
@@ -880,85 +912,95 @@ && isPersistentReferenceField(path.get(0))
   private String createExistsFilterHQL(int index, Filter filter, List<Property> path) {
     Property compared = path.get(path.size() - 1);
     Property collection = path.get(path.size() - 2);
-    String tableName = "ft_" + index;
+    Class<?> table = collection.getItemKlass();
+    String alias = alias(table, index);
     String pathToCollection =
         path.size() == 2
             ? path.get(0).getFieldName()
             : path.get(0).getFieldName() + "." + path.get(1).getFieldName();
-    return String.format(
-        "exists (select 1 from %2$s %1$s where %1$s in elements(e.%3$s) and %4$s)",
-        tableName,
-        collection.getItemKlass().getSimpleName(),
-        pathToCollection,
-        createFilterHQL(index, filter, tableName + "." + compared.getFieldName()));
+    Map<String, String> variables =
+        Map.ofEntries(
+            entry("table", table.getSimpleName()),
+            entry("alias", alias),
+            entry("path", pathToCollection),
+            entry("filter", createFilterHQL(index, filter, alias + "." + compared.getFieldName())));
+    return replace(
+        "exists (select 1 from ${table} ${alias} where ${alias} in elements(e.${path}) and ${filter})",
+        variables);
   }
 
   private String createFilterHQL(int index, Filter filter, String field) {
     Comparison operator = filter.getOperator();
     if (operator.isAccessCompare()) {
       return createAccessFilterHQL(index, filter, field);
     }
-    StringBuilder str = new StringBuilder();
-    String fieldTemplate = "%s";
+    String template = "${property}";
     if (filter.isAttribute()) {
-      fieldTemplate = "jsonb_extract_path_text(%s, '" + filter.getPropertyPath() + "', 'value')";
+      template =
+          "jsonb_extract_path_text(${property}, '" + filter.getPropertyPath() + "', 'value')";
     } else {
       Property property = context.resolveMandatory(filter.getPropertyPath());
       if (property.getPropertyType() == PropertyType.PASSWORD)
         throw new IllegalQueryException("Filter not allowed: " + filter);
       if (isStringLengthFilter(filter, property)) {
-        fieldTemplate = "length(%s)";
+        template = "length(${property})";
       } else if (isCollectionSizeFilter(filter, property)) {
-        fieldTemplate = "size(%s)";
+        template = "size(${property})";
       } else if (isJsonCollectionFilter(filter, property)) {
-        return createJsonFilterHQL(index, filter, field);
+        return createJsonCollectionFilterHQL(filter, field);
       } else if (operator.isCaseInsensitive()) {
-        fieldTemplate = "lower(%s)";
+        template = "lower(${property})";
       }
     }
-    str.append(String.format(fieldTemplate, field));
-    str.append(" ").append(createOperatorLeftSideHQL(operator));
-    if (!operator.isUnary()) {
-      str.append(" :f_").append(index).append(createOperatorRightSideHQL(operator));
-    }
-    return str.toString();
```

Comment:
FYI: This was super hard to understand because the HQL for the operator and their arguments was so split apart. That is why I moved the SQL/HQL of just the operator itself into the `Operator` enum but everything beyond is now in the template and the variables.